src/battle_ai_util.c

```diff
@@ -2752,8 +2752,18 @@ bool32 AI_CanPutToSleep(u32 battlerAtk, u32 battlerDef, u32 defAbility, u32 move
 
 static bool32 AI_CanPoisonType(u32 battlerAttacker, u32 battlerTarget, u32 move)
 {
-    return ((AI_DATA->abilities[battlerAttacker] == ABILITY_CORROSION && gBattleMoves[move].split == SPLIT_STATUS)
-            || !(IS_BATTLER_OF_TYPE(battlerTarget, TYPE_POISON) || IS_BATTLER_OF_TYPE(battlerTarget, TYPE_STEEL)));
+    if (GetBattlerAbility(battlerAttacker) == ABILITY_CORROSION)
+    {
+        if ((IS_BATTLER_OF_TYPE(battlerTarget, TYPE_STEEL) && (gBattleMoves[gCurrentMove].split == SPLIT_STATUS))
```

Comment:
First off:
```suggestion
        if ((IS_BATTLER_OF_TYPE(battlerTarget, TYPE_STEEL) && (gBattleMoves[move].split == SPLIT_STATUS))
```

Second, this isn't a good way to handle it. The AI should think it can poison a Steel type or Poison type with any move. If the engine is coded correctly, the effectiveness check will take over and the move won't even be considered because it doesn't affect the target (for damaging Poison-type moves on Steel-types). Additionally, if Inverse battles are ever added to the expansion, this kind of logic (of only status moves) will cause AI judgment errors because it won't think it could poison a Steel-type even though it could.

Third, AI_CanPoisonType should be just calling `CanPoisonType`. Or better yet, remove the function altogether and just call `CanPoisonType`.

Reply:
> Third, AI_CanPoisonType should be just calling CanPoisonType. Or better yet, remove the function altogether and just call CanPoisonType.

I don't know why but the expansion doesn't pass a move argument in the battle_script_command functions so since I can't tell why the decision was made to make additional AI functions I don't want to make a change. As suggested for now I will just change it to call `CanPoisonType` since the argument is not needed anymore.

> The AI should think it can poison a Steel type or Poison type with any move. If the engine is coded correctly, the effectiveness check will take over and the move won't even be considered because it doesn't affect the target

I see.

> Additionally, if Inverse battles are ever added to the expansion, this kind of logic (of only status moves) will cause AI judgment errors because it won't think it could poison a Steel-type even though it could.

Inverse battles are already a feature. 

Pushed a change.